app/lib/synthetic_branding_group_populator.rb

```diff
@@ -0,0 +1,86 @@
+##
+# SyntheticBrandingGroupPopulator: class to populate branding groups using the output of SyntheticStudyPopulator
+##
+
+class SyntheticBrandingGroupPopulator
+  BASE_BRANDING_GROUPS_CONFIG_PATH = Rails.root.join('db', 'seed', 'synthetic_branding_groups')
+
+  # populate all branding groups using synthetic studies
+  #
+  # @param user (User)                 => User account to own the branding groups
+  # @param create_studies (Boolean)    => Set to true to run SytheticStudyPopulator to create/overwrite studies
+  def self.populate_all(user: User.first, create_studies: false)
+    # populate all studies first to ensure they exist, if specified
+    SyntheticStudyPopulator.populate_all if create_studies
+    synthetic_studies_ids = SyntheticStudyPopulator.collect_synthetic_studies.pluck(:id)
+    configurations = self.load_all_config_files
+    configurations.each do |configuration|
+      study_regex = /#{configuration.dig('study_title_regex')}/i
+      study_list = Study.where(name: study_regex, :id.in => synthetic_studies_ids)
```

Comment:
let's move these two lines into the populate method.  So the division of concerns is populate_all understands how to find all the configuration objects, and populate understands how to take a config object and do everything needed to populate it

Reply:
Gotcha

Reply:
6ba85a7